src/identity/IdentityProviderFactory.ts

```diff
@@ -0,0 +1,77 @@
+import type { AnyObject,
+  CanBePromise,
+  interactionPolicy as InteractionPolicy,
+  KoaContextWithOIDC,
+  Configuration,
+  Account } from 'oidc-provider';
+import { Provider } from 'oidc-provider';
+
+import type { IdPConfigurationFactory } from './configuration/IdPConfigurationFactory';
+
+export class IdentityProviderFactory {
+  private readonly issuer: string;
+  private readonly configurationFactory: IdPConfigurationFactory;
+
+  public constructor(
+    issuer: string,
+    configurationFactory: IdPConfigurationFactory,
+  ) {
+    this.issuer = issuer;
+    this.configurationFactory = configurationFactory;
+  }
+
+  public createProvider(interactionPolicyOptions: {
+    policy?: InteractionPolicy.Prompt[];
+    url?: (ctx: KoaContextWithOIDC) => CanBePromise<string>;
```

Comment:
What is the impact of Koa on our code? Need to describe that in comments.

Reply:
The impacts are best described in the comments in `IdentityProviderHttpHandler`. panva/node-oidc-provider is built with Koa at its core, but I tried my best to remove as many references to it as I possibly could. Koa is no longer a direct dependency. It is only indirectly included through node-oidc-provider